posthog/warehouse/api/saved_query.py

```diff
@@ -190,25 +190,25 @@ def descendants(self, request: request.Request, *args, **kwargs) -> response.Res
         look further ahead into the descendants tree. If `level` overshoots (i.e. points to only
         descendants further than a leaf), we return an empty list.
         """
-        level = request.data.get("level", 1)
+        up_to_level = request.data.get("level", None)
 
         saved_query = self.get_object()
         saved_query_id = saved_query.id.hex
 
-        lquery = f"*.{saved_query_id}.*{{{level},}}"
+        lquery = f"*.{saved_query_id}.*{{1,}}"
         paths = DataWarehouseModelPath.objects.filter(team=saved_query.team, path__lquery=lquery)
 
         if not paths:
             return response.Response({"descendants": []})
 
-        descendants = set()
-
+        descendants: set[str] = set()
         for model_path in paths:
-            offset = model_path.path.index(saved_query_id) + level
-
-            if offset > len(model_path.path):
-                continue
+            start = model_path.path.index(saved_query_id) + 1
```

Comment:
In a world without the ORM we could probably call `subpath` in pg to get a path starting at `saved_query_id`, and then just use `0` here as the start.

Reply:
Hell, in a world without the ORM the endpoint could just be a query probably.